src/palimpzest/execution/execution_engine.py

```diff
@@ -0,0 +1,333 @@
+        )
+
+        # return the output records and plan stats
+        return records, [plan_stats]
+
+
+    def execute_confidence_interval_strategy(
+            self,
+            dataset: Set,
+            optimizer: Optimizer,
+            execution_data: List[RecordOpStats]=[],
+        ) -> Tuple[List[DataRecord], List[PlanStats]]:
+        # initialize output records and plan stats
+        records, plan_stats = [], []
+
+        # get the initial set of optimal plans according to the optimizer
+        plans = optimizer.optimize(dataset)
+        still_processing = True
+        while len(plans) > 1 and still_processing:
+            # identify the plan with the highest quality in the set
+            max_quality_plan_id = self.get_max_quality_plan_id(plans)
+
+            # execute the set of plans for a fixed number of samples
+            new_execution_data, new_records, new_plan_stats = self.execute_plans(list(plans), max_quality_plan_id, self.num_samples)
+            records.extend(new_records)
+            plan_stats.extend(new_plan_stats)
+
+            # set still processing to False if there are no new records
+            if new_records == []:
+                still_processing = False
+
+            if still_processing:
+                # update cost model and optimizer
+                execution_data.extend(new_execution_data)
+                cost_model = CostModel(
+                    source_dataset_id=self.source_dataset_id,
+                    sample_execution_data=execution_data,
+                )
+                optimizer.update_cost_model(cost_model)
+
+                # get new set of plans
+                plans = optimizer.optimize(dataset)
+
+        # execute final plan until end
+        final_plan = plans[0]
+        new_records, new_plan_stats = self.execute_plan(
+            plan=final_plan,
+            max_workers=self.max_workers,
+        )
+        records.extend(new_records)
+        plan_stats.append(new_plan_stats)
+
+        # return the final set of records and plan stats
+        return records, plan_stats
+
+
+    def execute_pareto_optimal_strategy(
+            self,
+            dataset: Set,
+            optimizer: Optimizer,
+            execution_data: List[RecordOpStats]=[],
+        ) -> Tuple[List[DataRecord], List[PlanStats]]:
+        # initialize output records and plan stats
+        records, plan_stats = [], []
+
+        # get the initial set of optimal plans according to the optimizer
+        plans = optimizer.optimize(dataset)
+        plan_set, prev_plan_set = set([plan.plan_id for plan in plans]), None
+        trials = 1
+        still_processing = True
+        while (
+            (trials < self.pareto_strategy_min_trials or plan_set != prev_plan_set)
+            and still_processing
+        ):
+            # identify the plan with the highest quality in the set
+            max_quality_plan_id = self.get_max_quality_plan_id(plans)
```

Comment:
Either this should be plan_set, or could be moved out of the while loop right?

Reply:
Or maybe I am a bit confused by the structure of the loop. We are always iterating on `plans` but keeping track of `plan_set` for the termination condition?
I think it would be good to add a short docstring of this function for future reference

Reply:
Ended up scrapping this function entirely as this seems to be an optimization strategy which is not really useful